### PR TITLE
feat: adding test cases

### PR DESCRIPTION
**Pull request by joker00777**

### CODE REVIEW

**Comment by ankitm123** [comment]
Very nice, I will take a look this week

File: go.mod [hunk 1]
```diff
@@ -23,6 +23,7 @@ require (
 	github.com/tektoncd/pipeline v0.32.4
 	gocloud.dev v0.25.0
 	golang.org/x/text v0.3.7
+	golang.org/x/tools v0.1.12
 	k8s.io/api v0.22.15
```
Comment (ankitm123) [inline go.mod:27]: Why this change?

Comment (joker00777) [reply go.mod]: not needed.

File: pkg/cloud/buckets/buckets.go [hunk 2]
```diff
@@ -26,6 +26,14 @@ import (
 	_ "gocloud.dev/blob/s3blob"
 )
 
+type HTTPClient interface {
```
Comment (ankitm123) [inline pkg/cloud/buckets/buckets.go:29]: More clarity, lets put this in an interface.go file

File: pkg/cloud/buckets/buckets.go [hunk 3]
```diff
@@ -26,6 +26,14 @@ import (
...
+	Do(*http.Request) (*http.Response, error)
+}
+
+var (
```
Comment (ankitm123) [inline pkg/cloud/buckets/buckets.go:33]: will be flagged by the linter, no need for the brackets.

File: pkg/cloud/buckets/buckets_test.go [hunk 4]
```diff
@@ -27,3 +46,179 @@ func assertSplitBucketURL(t *testing.T, inputURL, expectedBucketURL, expectedKey
...
+		err = buckets.WriteBucketURL(ctx, u, strings.NewReader(tc.data), tc.time)
+		if tc.expectedError != "" {
+			assert.Contains(t, err.Error(), "failed to open bucket")
+		} else {
+			assert.NoError(t, err)
+		}
+		read, err := buckets.ReadURL(ctx, tc.url, tc.time, nil)
+		if tc.expectedError != "" {
+			assert.Nil(t, read)
+			assert.Contains(t, err.Error(), "failed to open bucket")
```
Comment (ankitm123) [inline pkg/cloud/buckets/buckets_test.go:216]: I think it's enough to test that there is an error, using contains is a bit brittle here.

**Comment by msvticket** [comment]
This PR is clearly abandoned by the author
